docs/DevGuide/Databox.md

```diff
@@ -0,0 +1,367 @@
+
+What changes as our program's size increases in scale? In SpECTRE, one of our
+driving design goals is modularity. That is, that functionality can be easily
+swapped in and out as desired. These smaller modules are easier to test, and
+keep the code flexible. We could wrap our calculation in such a module, which
+would then allow us to decouple the initial setup of the variables from where
+they are used in calculations:
+
+\snippet Test_DataBoxDocumentation.cpp working_without_databoxes_mass_compute
+
+\snippet Test_DataBoxDocumentation.cpp working_without_databoxes_accel_compute
+
+Our small program can now be written as:
+
+\snippet Test_DataBoxDocumentation.cpp working_without_databoxes_small_program_2
+
+One advantage is immediate: we are free to add other computation modules that
+are independently testable and reusable. As the number of routines grows, we
+can even begin to write routines that work on top of existing ones.
+
+\snippet Test_DataBoxDocumentation.cpp working_without_databoxes_force_compute
+
+While we have made progress, two problems arise. Our first problem is that as
+the number of quantities grows, the more unwieldy it becomes to have to specify
+each function argument in our routine. The second problem is worse: the
+arguments passed to functions can be transposed and the program will compile
+and run just the same as before. That is, the following two lines are equally
+well-formed from the point of view of the program:
+
+\snippet Test_DataBoxDocumentation.cpp working_without_databoxes_failed_accel
+
+Every time we call `acceleration_compute` we need to make sure we pass in the
+arguments in the correct order. For large programs where `acceleration_compute`
+is called many times, it becomes inevitable that the arguments will be
+accidentally transposed. We can address the two problems described above with a
+`std::map`, the first container we'll consider in this series.
+
+##A std::map DataBox {#first_attempt}
+We can encapsulate the variables we use in a `std::map`, and the first half of
+our small program example now looks like this:
+
+\snippet Test_DataBoxDocumentation.cpp std_map_databox_small_program_1
+
+We have not yet taken full advantage of the encapsulation that `std::map`
+provides. We do so by rewriting our other routines to only take a single
+argument: the `std::map` itself:
+
+\snippet Test_DataBoxDocumentation.cpp std_map_databox_mass_compute
+
+\snippet Test_DataBoxDocumentation.cpp std_map_databox_accel_compute
+
+Notice that this solves the problem of having to provide the arguments
+in the correct order to every function call of `acceleration_compute`:
+
+\snippet Test_DataBoxDocumentation.cpp std_map_databox_force_compute
+
+
+Our small program now looks like:
+
+\snippet Test_DataBoxDocumentation.cpp std_map_databox_small_program_2
+
+We could even replace the last few lines with one:
+
+~~~{.c}
+return force_compute(naive_databox);
```

Comment:
Move force_compute into the larger code block, and don't use mass_compute and acceleration_compute